helix-tui/src/text.rs

```diff
@@ -229,6 +229,70 @@ impl<'a> Spans<'a> {
     pub fn width(&self) -> usize {
         self.0.iter().map(Span::width).sum()
     }
+
+    /// Constructs a styled `Span` from the given `string` and `style`
+    /// and appends it the to the `Spans`.
+    ///
+    /// ## Examples
+    ///
+    /// ```rust
+    /// # use helix_tui::text::{Span, Spans};
+    /// # use helix_view::graphics::{Color, Style};
+    /// let mut spans = Spans::new();
+    /// spans.push("Hello!", Style::default());
+    /// assert_eq!("Hello!", spans.0[0].content);
+    /// ```
+    pub fn push<T>(&mut self, string: T, style: Style)
```

Comment:
This should take a `Span<'a>` instead

Reply:
It makes sense, but that will bloat the current usages a bit. I guess I could always add a helper function there instead. Will change and see if a utility function becomes necessary in `statusline.rs`.

Reply:
Fixed in 8e8d860b. Boilerplate is minimal, no extra helper added.